Snowflake.Data/Client/SnowflakeDbConnection.cs

```diff
@@ -24,8 +26,155 @@ public class SnowflakeDbConnection : DbConnection
 
         internal int _connectionTimeout;
 
+        private bool isActive = false;
+
         private bool disposed = false;
 
+        /// <summary>
+        ///     Connection pool 
+        /// </summary>
+        private static ConcurrentDictionary<Tuple<string, SecureString>, SnowflakeDbConnection> connectionPool;

```

Comment:
Do not feel it is a good idea to have connection pool info and logic inside SnowflakeDbConnection. There should be a ConnectionPool to wrap and hide the pool related logic. Such as https://github.com/mysql-net/MySqlConnector/blob/master/src/MySqlConnector/Core/ConnectionPool.cs

Reply:
As long as the connection pool is hidden from me, the user of the DbConnection class, I couldn't care less where the logic lives.

I just need to know that pooling is happening somewhere so that we don't leak http clients.